pkg/util/names/names.go

```diff
@@ -121,3 +121,19 @@ func IsReservedNamespace(namespace string) bool {
 func GenerateImpersonationSecretName(clusterName string) string {
 	return fmt.Sprintf("%s-impersonator", clusterName)
 }
+
+// GeneratePropagationPolicyName generates the propagationPolicy name
+func GeneratePropagationPolicyName(namespace, name, gvk string) string {
+	hash := fnv.New32a()
+	hashutil.DeepHashObject(hash, gvk)
+
+	return strings.ToLower(fmt.Sprintf("%s-%s-%s", namespace, name, rand.SafeEncodeString(fmt.Sprint(hash.Sum32()))))
+}
+
+// GenerateClusterPropagationPolicyName generates the clusterPropagationPolicy name
+func GenerateClusterPropagationPolicyName(name, gvk string) string {
+	hash := fnv.New32a()
+	hashutil.DeepHashObject(hash, gvk)
+
+	return strings.ToLower(fmt.Sprintf("%s-%s", name, rand.SafeEncodeString(fmt.Sprint(hash.Sum32()))))
+}
```

Comment:
Could we unify `GeneratePropagationPolicyName` and `GenerateClusterPropagationPolicyName`, like `GenerateWorkName`?

directly use `GeneratePolicyName`? I consider that we could possiblly use `OverridePolicy` that has this problem in future.